packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -287,13 +294,449 @@ class HeaderMatchImpl extends HttpHeaderMatch {
+   */
+  static valueIs(queryParameterName: string, queryParameterValue: string): QueryParameterMatch {
+    return new QueryParameterMatchImpl(queryParameterName, { exact: queryParameterValue });
+  }
+
+  /**
+   * Returns the query parameter match configuration.
+   */
+  abstract bind(scope: Construct): QueryParameterConfig;
+}
+
+class QueryParameterMatchImpl extends QueryParameterMatch {
+  constructor(
+    private readonly queryParameterName: string,
+    private readonly matchProperty: CfnRoute.HttpQueryParameterMatchProperty,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): QueryParameterConfig {
+    return {
+      queryParameter: {
+        match: this.matchProperty,
+        name: this.queryParameterName,
+      },
+    };
+  }
+}
+
+/**
+ * Configuration for gRPC route match.
+ */
+export interface GrpcRouteMatchConfig {
+  /**
+   * Route CFN configuration for gRPC route match.
+   */
+  readonly requestMatch: CfnRoute.GrpcRouteMatchProperty;
+}
+
+/**
+ * Other optional properties for gRPC route match other than the service name.
  */
-export interface GrpcRouteMatch {
+export interface GrpcRouteServiceNameMatchOptions {
   /**
-   * The fully qualified domain name for the service to match from the request
+   * The route metadata to be matched on.
+   *
+   * @default - no match on metadata
+   *
+   */
+  readonly methodName?: string;
+
+  /**
+   * The route metadata to be matched on.
+   *
+   * @default - no match on metadata
+   *
+   */
+  readonly metadata?: GrpcMetadataMatch[];
+}
+
+/**
+ * Optional properties for gRPC route match other than the metadata.
+ */
+export interface GrpcRouteMetadataMatchOptions {
+  /**
+   * The fully qualified domain name for the service to match from the request.
+   *
+   * @default - no match on service name
    */
   readonly serviceName: string;
+
+  /**
+   * The route metadata to be matched on.
+   *
+   * @default - no match on metadata
+   *
+   */
+  readonly methodName?: string;
+}
+
+/**
+ * The criterion for determining a request match for this Route.
+ */
+export abstract class GrpcRouteMatch {
+  /**
+   * Create service name based gRPC route match.
+   *
+   * @param serviceName The fully qualified domain name for the service to match from the request
+   * @param options Optional properties to define gRPC route match
+   */
+  public static serviceName(serviceName: string, options?: GrpcRouteServiceNameMatchOptions): GrpcRouteMatch {
+    return new GrpcRouteServicenameMatchImpl(serviceName, options);
+  }
+
+  /**
+   * Create metadata based gRPC route match.
+   *
+   * @param metadata An object that represents the data to match from the request.
+   * @param options Optional properties to define gRPC route match
+   */
+  public static metadata(metadata: GrpcMetadataMatch[], options?: GrpcRouteMetadataMatchOptions): GrpcRouteMatch {
+    return new GrpcRouteMetadataMatchImpl(metadata, options);
+  }
```

Comment:
Same comment as with `GrpcGatewayRouteMatch` before - this doesn't really make sense as a union-like class. Please replace it with an interface.

Reply:
Well noted.